Dockerfile

```diff
@@ -0,0 +1,23 @@
+FROM python:3.7 as app
+
+WORKDIR /usr/src/app
+# Install required dependencies and cache test dependencies, so they won't be redownloaded when updating after the code
+# is changed. (We cannot use setup.py here, since it requires other files from this package.)
+COPY requirements.txt requirements-tests.txt ./
+RUN pip install -r requirements.txt
+RUN pip download -r requirements-tests.txt
+COPY . .
+RUN pip install .
```

Comment:
isn't `pip install .` running a setup.py which requires other files from this package?

Reply:
Yes, it's the reason for `COPY` above.